kratos/utilities/condition_number_utility.h

```diff
@@ -84,27 +87,44 @@ class ConditionNumberUtility
     ///@{
 
     /// Default constructor.
-    ConditionNumberUtility() 
-    {
-        
-    }
+    ConditionNumberUtility()
+        :mpEigenSolverMax(nullptr), 
```

Comment:
Do you really need a default constructor? I would avoid it as it puts the utility uncomplete....

Reply:
It is useful to provide an interface to set manually the solvers once the utility has been created

Reply:
if you are not using it now just delete it. (I would avoid its use anyway) You are breaking two rules:
1. Don't over design or Only write code if you absolutely have to
2. Don't let users create an invalid object 

You will implement it when you need it but now it would be a source of uncertainty. (Consider this as a general coding recommendation)

Reply:
I am actually using it, but in a another branch